ada_feeding_perception/ada_feeding_perception/republisher.py

```diff
@@ -171,10 +380,158 @@ def callback(msg: Any):
             # self.get_logger().info(
             #     f"Received message on topic {i} {self.from_topics[i]}"
             # )
+            for post_processor in post_processors:
+                msg = post_processor(msg)
             self.pubs[i].publish(msg)
 
         return callback
 
+    def mask_post_processor(self, msg: Image) -> Image:
```

Comment:
Wow, this is machinery to get this to work, kudos!

For readability / modularity, consider putting post-processors in their own module.